pkg/scheduler/scheduler.go

```diff
@@ -237,16 +237,29 @@ func (s *Scheduler) FrontendLoop(frontend schedulerpb.SchedulerForFrontend_Front
 
 		switch msg.GetType() {
 		case schedulerpb.ENQUEUE:
-			err = s.enqueueRequest(frontendCtx, frontendAddress, msg)
+			// Extract tracing information from headers in HTTP request. FrontendContext doesn't have the correct tracing
+			// information, since that is a long-running request.
+			tracer := opentracing.GlobalTracer()
+			parentSpanContext, err := httpgrpcutil.GetParentSpanForRequest(tracer, msg.HttpRequest)
+			if err != nil {
+				return err
+			}
+
+			enqueueSpan, _ := opentracing.StartSpanFromContextWithTracer(frontendCtx, tracer, "enqueue", opentracing.ChildOf(parentSpanContext))
```

Comment:
`enqueueSpan` already has `frontendCtx` as it's parent, do we need to pass both `frontendCtx` and `enqueueSpan.Context()` to `s.enqueueRequest`?

Reply:
comment in the original code says that frontendCtx is of the long running loop, but we want to create the span context from the current request headers, so no

Reply:
I wasn't clear. My suggestion was to only pass `enqueueSpan.Context()` to `s.enqueueRequest`.

Reply:
enqueueSpan.Context() is a SpanContext. Not sure how to use it as go Context in the called function

Reply:
You're correct. What I had in mind is `context.Context` returned as a second value from `opentracing.StartSpanFromContextWithTracer`, which is currently ignored (`_`).

Reply:
> enqueueSpan.Context() is a SpanContext. Not sure how to use it as go Context in the called function

Ah, I see what you mean.

Instead of calling `opentracing.StartSpanFromContext(ctx, "queued", opentracing.ChildOf(parentSpanContext))` inside `enqueueRequest`, we could then call `opentracing.StartSpanFromContext(ctx, "queued")`, because parent span context is already stored in `ctx` returned by `opentracing.StartSpanFromContextWithTracer` when creating `enqueueSpan`.

Then we wouldn't set `req.parentSpanContext` inside `enqueueRequest` -- that's only done for a test.

Reply:
updated